code/mob/living/silicon/ai.dm

```diff
@@ -659,6 +659,12 @@ var/list/ai_emotions = list("Happy" = "ai_happy",\
 		//Might as well log what they said too!
 		logTheThing("diary", src, null, ": [t]", "say")
 
+	if (href_list["net_id"]) // this is triggered by examine() in atom.dm, though you should be able to use this from elsewhere!
+		var/id = href_list["net_id"]
+		var/owner = href_list["owner"]
+		var/message = input(usr, "Please enter message", owner, null)
```

Comment:
consider tgui input

Reply:
Currently PDAs use href links and this form of input for sending texts afaik; are you sure? (Would have to take a while to learn TGUI but I can definitely try!)

Reply:
No, consider `tgui_input()` for this line.

Reply:
Ah okay!